doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+; field = value
+vni = 1*8DIGIT    ; vni to be used for this VLAN when sending traffic to the remote VTEP
+```
+
+**VXLAN_FDB_TABLE**
+
+Producer: fdbsyncd
+
+Consumer: fdborch
+
+Description: This holds the MACs learnt over a VNI from the remote VTEP.  This is populated by fdbsyncd on receipt of a MAC route. 
+
+Schema:
+
+```
+; New table
+; specifies the MAC and VLAN to be programmed over a tunnel
+
+key = VXLAN_FDB_TABLE:"Vlan"vlanid:mac_address
+                          ;MAC Address and VLAN ID
+;field = value
+remote_vtep = ipv4
+type          = "static" / "dynamic"  ; type of the entry.
+vni         = 1*8DIGIT                ; vni to be used for this VLAN when sending traffic to the remote VTEP
+```
+
+**VRF_TABLE**
+
+Producer: VrfMgr
+
+Consumer: Vrforch
+
+Description: Updated existing table to store VNI associated with VRF.
+
+Schema:
+
+```
+;Existing table
+;defines virtual routing forward table. Updated to stores VNI associated with VRF
+;
+;Status: stable
+
+key = VRF_TABLE:VRF_NAME ;
+fallback = "true"/"false"             ; default false
+vni = 1*8DIGIT ; VNI assicated with the VRF
+
+```
+
+**VRF_ROUTE_TABLE**
+
+Producer: fpmsyncd
+
+Consumer: routeorch
+
+Description: VRF Route Table is extended for Tunnel Nexthop. 
+
+Schema:
+
+```
+;Stores a list of routes
+;Status: Mandatory
+
+key = ROUTE_TABLE:VRF_NAME:prefix ;
+nexthop = prefix,              ; IP addresses separated ',' (empty indicates no gateway). May indicate VXLAN endpoint if vni_label is non zero.
+intf = ifindex? PORT_TABLE.key ; zero or more separated by ',' (zero indicates no interface)
+vni_label = VRF.vni            ; New Field. zero or more separated by ',' (empty value for non-vxlan next-hops). May carry MPLS label in future.
+router_mac = mac_address       ; New Field. zero or more remote router MAC address separated by ',' (empty value for non-vxlan next-hops)
+blackhole = BIT                ; Set to 1 if this route is a blackhole (or null0)
+
+```
+
+**NEIGH_SUPPRESS_APP_VLAN_TABLE**
+
+Producer: VlanMgr
+
+Consumer: VlanOrch
+
+Description: New table for per VLAN neighbor suppression configuration.
+
+Schema:
+
+```
+; New table
+;Stores Neigh Suppress configuration per VLAN
+
+key             = NEIGH_SUPPRESS_VLAN:"Vlan"vlanid		; VLAN with prefix "NEIGH_SUPPRESS"
+suppress_neigh  = "ON" / "OFF" ; Default "OFF"
+```
+
+### 4.2.3 STATE_DB changes
+
+**STATE_VXLAN_TUNNEL_TABLE**
+
+Producer: vxlanorch
+
+Consumer: show scripts
+
+Description: 
+
+- Holds the created tunnels - static or dynamic. 
+- There is a VXLAN Tunnel entry for every tunnel created. 
+- The SIP and DIP are added for tunnels. 
+- Tunnel Creation source is added. 
+
+Schema:
+
+```
+; New table
+; Defines schema for VXLAN State
+key             = VXLAN_TUNNEL:name    ; tunnel name
+; field         = value
+SRC_IP          = ipv4                 ; tunnel sip
+DST_IP          = ipv4                 ; tunnel dip 
+tnl_src         = "CLI"/"EVPN"  
```

Comment:
the option "CLI" is for static VXLAN case? There is a mention of not supporting static VXLAN in this document.

Reply:
Yes for the static case or could be extended for orchestrator configured tunnels like NSX. This is purely informational. With the current feature only EVPN will be filled in. Static tunnels configured through VXLAN_TUNNEL table with a DIP could fill this with the tnl_src field as static.